tests/Resources/Page/ODH/JupyterHub/JupyterLabLauncher.robot

```diff
@@ -235,11 +236,23 @@ Wait Until JupyterLab Is Loaded
   Wait Until Element Is Visible  xpath:${JL_TABBAR_CONTENT_XPATH}  timeout=${timeout}
 
 Clone Git Repository
+  [Documentation]    Clones git repository and logs error message if fails to clone
   [Arguments]  ${REPO_URL}
-  Navigate Home (Root folder) In JupyterLab Sidebar File Browser
-  Open With JupyterLab Menu  Git  Clone a Repository
-  Input Text  //div[.="Clone a repo"]/../div[contains(@class, "jp-Dialog-body")]//input  ${REPO_URL}
-  Click Element  xpath://div[.="CLONE"]
+  ${status}    ${err_msg} =    Run Keyword and Ignore Error    Clone Repo and Return Error Message    ${REPO_URL}
+    IF    "${status}" == "PASS"
+        ${dir_name} =    Get Directory Name From Git Repo URL    ${REPO_URL}
+        Delete Folder In User Notebook
+        ...    admin_username=${OCP_ADMIN_USER.USERNAME}
+        ...    username=${TEST_USER.USERNAME}
+        ...    folder=${dir_name}
+        ${status}    ${err_msg} =    Run Keyword and Ignore Error    Clone Repo and Return Error Message    ${REPO_URL}
+        IF    "${status}" == "PASS"
+            Log    Error Message : ${err_msg}
+            FAIL
+        END
```

Comment:
could you explain what's the goal here?

If I got right, if the clone process fails, then it deletes the repo dir (_but was the directory created if the clone process failed?_) and then it tries to clone again. Is that right?